packages/hoppscotch-app/newstore/environments.ts

```diff
@@ -76,6 +77,26 @@ const dispatchers = defineDispatchers({
       ],
     }
   },
+  duplicateEnvironment(
+    { environments }: EnvironmentStore,
+    { envIndex }: { envIndex: number }
+  ) {
+    const newEnvironment = environments.find((_, index) => index === envIndex)
+    if (!newEnvironment) {
+      return {
+        environments,
+      }
+    }
+    return {
+      environments: [
+        ...environments,
+        {
+          ...cloneDeep(newEnvironment),
+          name: `${newEnvironment.name} - ${environments.length + 1}`,
```

Comment:
Appending `environments.length + 1` to duplicated entry's name doesn't feel right to me, instead, how about appending the incrementing count if the environment has entries with the same name.

Ex: Before duplicating `Foo`
Environments:
- Foo
- Bar
- Choco
- Baz

After duplicating `Foo`
Environments:
- Foo - 2  // instead of `Foo - 5`<-total environments.length+1
- Bar
- Choco
- Bax

Ideally the above should make more sense with duplicate's naming, or we can prefer something very simple like `Duplicate of <old environment name>` for the new entry's name.

Reply:
Thanks @liyasthomas Yes, good suggestions. I would prefer going with something very simple. I like you 2 option :)